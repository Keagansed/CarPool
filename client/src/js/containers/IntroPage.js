import React from "react";
import Slider from "react-slick";
import { Link } from 'react-router-dom';

import slide_1  from "../../css/images/Intro_0.png";
import slide_2  from "../../css/images/Intro_1.png";
import slide_3  from "../../css/images/Intro_2.png";
import slide_4  from "../../css/images/Intro_3.png";

class IntroPage extends React.Component {
    render() {
        var settings = {
            dots: false,
            arrows: false,
            infinite: false,
            speed: 500,
            slidesToShow: 1,
            slidesToScroll: 1
        };
        return (
            <div className="bg-purple vertical-center">
                <div className="container-fluid pad-0">
                    <Slider {...settings} className="size-100">
                        <div>
                            <img className="img-fluid d-block mx-auto size-100" src={slide_1} alt="slideOne"/>
                        </div>
                        <div>
                            <img className="img-fluid d-block mx-auto size-100" src={slide_2} alt="slideTwo"/>
                        </div>
                        <div>
                            <img className="img-fluid d-block mx-auto size-100" src={slide_3} alt="slideThree"/>
                        </div>
                        <div>
                            <img className="img-fluid d-block mx-auto size-100" src={slide_4} alt="slideFour"/>
                        </div>
                        <div className="txt-center">
                            <p className="txt-center txt-white mx-auto">Now that you now how the Carpool Platform works, all you need to do is log into your account to get started!</p>
                            <Link to={`/Login`} className="btn btn-primary mx-auto width-15rem brad-2rem bg-aqua txt-purple fw-bold" id="btnLoginPage">
                                Proceed to Login
                            </Link>
                        </div>
                    </Slider> 
                </div>
            </div>
        );
    }
}

export default IntroPage;