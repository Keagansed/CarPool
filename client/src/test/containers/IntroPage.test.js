import React from 'react';
import IntroPage from '../../js/containers/IntroPage';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';

describe('Landing Page Component', () => {

    it ('captures snapshot', () => {
        const renderedValue = renderer.create(<MemoryRouter><IntroPage /></MemoryRouter>).toJSON()
        expect(renderedValue).toMatchSnapshot();
    });

    it ('renders correctly', () => {
        let wrapper = shallow(<MemoryRouter><IntroPage /></MemoryRouter>)
        expect(wrapper.length).toEqual(1);
    });
});