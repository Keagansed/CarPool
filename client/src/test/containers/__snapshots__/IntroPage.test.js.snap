// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IntroPage Component captures snapshot 1`] = `
<div
  className="bg-purple vertical-center"
>
  <div
    className="container-fluid pad-0"
  >
    <div
      className="slick-slider size-100 slick-initialized"
      dir="ltr"
    >
      <div
        className="slick-list"
        onClick={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={null}
        onMouseMove={null}
        onMouseUp={[Function]}
        onTouchCancel={null}
        onTouchEnd={[Function]}
        onTouchMove={null}
        onTouchStart={[Function]}
        style={Object {}}
      >
        <div
          className="slick-track"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          style={
            Object {
              "WebkitTransform": "translate3d(0px, 0px, 0px)",
              "WebkitTransition": "",
              "msTransform": "translateX(0px)",
              "opacity": 1,
              "transform": "translate3d(0px, 0px, 0px)",
              "transition": "",
            }
          }
        >
          <div
            aria-hidden={false}
            className="slick-slide slick-active slick-current"
            data-index={0}
            onClick={[Function]}
            style={
              Object {
                "outline": "none",
                "width": 0,
              }
            }
            tabIndex="-1"
          >
            <div>
              <div
                style={
                  Object {
                    "display": "inline-block",
                    "width": "100%",
                  }
                }
                tabIndex={-1}
              >
                <img
                  alt="slideOne"
                  className="img-fluid d-block mx-auto size-100"
                  src="Intro_0.png"
                />
              </div>
            </div>
          </div>
          <div
            aria-hidden={true}
            className="slick-slide"
            data-index={1}
            onClick={[Function]}
            style={
              Object {
                "outline": "none",
                "width": 0,
              }
            }
            tabIndex="-1"
          >
            <div>
              <div
                style={
                  Object {
                    "display": "inline-block",
                    "width": "100%",
                  }
                }
                tabIndex={-1}
              >
                <img
                  alt="slideTwo"
                  className="img-fluid d-block mx-auto size-100"
                  src="Intro_1.png"
                />
              </div>
            </div>
          </div>
          <div
            aria-hidden={true}
            className="slick-slide"
            data-index={2}
            onClick={[Function]}
            style={
              Object {
                "outline": "none",
                "width": 0,
              }
            }
            tabIndex="-1"
          >
            <div>
              <div
                style={
                  Object {
                    "display": "inline-block",
                    "width": "100%",
                  }
                }
                tabIndex={-1}
              >
                <img
                  alt="slideThree"
                  className="img-fluid d-block mx-auto size-100"
                  src="Intro_2.png"
                />
              </div>
            </div>
          </div>
          <div
            aria-hidden={true}
            className="slick-slide"
            data-index={3}
            onClick={[Function]}
            style={
              Object {
                "outline": "none",
                "width": 0,
              }
            }
            tabIndex="-1"
          >
            <div>
              <div
                style={
                  Object {
                    "display": "inline-block",
                    "width": "100%",
                  }
                }
                tabIndex={-1}
              >
                <img
                  alt="slideFour"
                  className="img-fluid d-block mx-auto size-100"
                  src="Intro_3.png"
                />
              </div>
            </div>
          </div>
          <div
            aria-hidden={true}
            className="slick-slide"
            data-index={4}
            onClick={[Function]}
            style={
              Object {
                "outline": "none",
                "width": 0,
              }
            }
            tabIndex="-1"
          >
            <div>
              <div
                style={
                  Object {
                    "display": "inline-block",
                    "width": "100%",
                  }
                }
                tabIndex={-1}
              >
                <img
                  alt="slideFive"
                  className="img-fluid d-block mx-auto size-100"
                  src="Intro_4.png"
                />
              </div>
            </div>
          </div>
          <div
            aria-hidden={true}
            className="slick-slide"
            data-index={5}
            onClick={[Function]}
            style={
              Object {
                "outline": "none",
                "width": 0,
              }
            }
            tabIndex="-1"
          >
            <div>
              <div
                style={
                  Object {
                    "display": "inline-block",
                    "width": "100%",
                  }
                }
                tabIndex={-1}
              >
                <img
                  alt="slideSix"
                  className="img-fluid d-block mx-auto size-100"
                  src="Intro_5.png"
                />
              </div>
            </div>
          </div>
          <div
            aria-hidden={true}
            className="slick-slide"
            data-index={6}
            onClick={[Function]}
            style={
              Object {
                "outline": "none",
                "width": 0,
              }
            }
            tabIndex="-1"
          >
            <div>
              <div
                style={
                  Object {
                    "display": "inline-block",
                    "width": "100%",
                  }
                }
                tabIndex={-1}
              >
                <img
                  alt="slideSeven"
                  className="img-fluid d-block mx-auto size-100"
                  src="Intro_6.png"
                />
              </div>
            </div>
          </div>
          <div
            aria-hidden={true}
            className="slick-slide"
            data-index={7}
            onClick={[Function]}
            style={
              Object {
                "outline": "none",
                "width": 0,
              }
            }
            tabIndex="-1"
          >
            <div>
              <div
                className="txt-center"
                style={
                  Object {
                    "display": "inline-block",
                    "width": "100%",
                  }
                }
                tabIndex={-1}
              >
                <div
                  className="vertical-center bg-purple"
                >
                  <div
                    className="container-fluid"
                  >
                    <div
                      className="row"
                    >
                      <img
                        alt="carpool_logo"
                        className="img-fluid d-block mx-auto mbottom-2rem"
                        id="logo-256"
                        src="logo.png"
                      />
                    </div>
                    <p
                      className="txt-center txt-white mx-auto"
                    >
                      Now that you now how the Carpool Platform works, all you need to do is log into your account to get started!
                    </p>
                    <div
                      className="row"
                    >
                      <a
                        className="btn btn-primary mx-auto width-15rem brad-2rem bg-aqua txt-purple fw-bold"
                        href="/Login"
                        id="btnLoginPage"
                        onClick={[Function]}
                      >
                        Proceed to Login
                      </a>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
